force-app/main/default/classes/CarTriggerUtility.cls

```diff
@@ -0,0 +1,63 @@
+public with sharing class CarTriggerUtility {
+    public static void afterInsert(List<Car__c> cars){
+        System.debug('utilitydebug');
+        List<Contact> listOfChangedContacts= new List<Contact>();
+        List<Contact> contacts = new List<Contact>();
+        
+        contacts= [SELECT Id,Expenses__c,LastAddedExpenseValue__c FROM Contact];
+
+        for(Car__c	 car: cars){
+            for(Contact contact: contacts){
+                if(car.Contact__c==contact.id){
+                    System.debug('Found Contact of Car');
+                    contact.LastAddedExpenseValue__c= car.Price__c;
+                    contact.Expenses__c+= car.Price__c;
+                    listOfChangedContacts.add(contact);
+                    }
+            }
+        }
```

Comment:
Generalnie całą to pętle można zrobić w troche prostrzy sposób i bez zapytania do bazy danych do wszystkich Contactów to znaczy. I pozwoli Ci to zrobić jedną metode pod update, insert i delete, ale metoda ta będzie odpowiedzialna za wywołanie pełnej rekalkulacji zawsze pod contactem ale będzie bezpieczna. 

Tzn. Najpierw lecisz pętlą po Carsach, żeby przechwycić sobie wszystkie Contacty jakie naruszone Carsy mają czyli...


metodka wyglądała by tak : 



Set<Id> contactIds = new Set<Id>();
for(Car__c car: cars) {
contactIds.add(car.Id);
}

//I teraz jak już masz Id wszystkich contactów to możesz zrobić taki mały zabieg, który zczytasz wszystkie Contacty i // Ich Cary czyli...

List<Contact__c> contactsWithCars = [SELECT Id, Expenses__c (SELECT Id, Price__c FROM Cars__r) FROM Contact WHERE Id IN: contactIds];

// i teraz możesz śmiało przelecieć pętlą i obliczyć na nowo wartości na contactie. Tzn:

List<Contact> contactsToUpdate= new List<Contact>();


for(Contact contactTemp : contactsWithCars) {
    Integer newContactExpenses = 0;
    for(Car__c carTemp : contactTemp.Cars__r) {
        newContactExpenses += carTemp.Expenses__c ;
    }
   // I teraz sprawdzasz dla updatu czy zmieniła się wartość contactu jaka była na początku a jaką Ty nadałeś, żeby 
   //stwierdzić czy updatowac

   if(newContactExpenses  != contactTemp.Expenses__c) {
        contactTemp.Expenses__c = newContactExpenses;
        contactsToUpdate.add(contactTemp);
   }
}
if(!contactsToUpdate.isEmpty()) {
    update contactsToUpdate;
}